group-coordinator/src/test/java/org/apache/kafka/coordinator/group/runtime/CoordinatorRuntimeTest.java

```diff
@@ -3839,6 +3839,114 @@ public void close() {}
         assertEquals("response1", write.get(5, TimeUnit.SECONDS));
     }
 
+    @Test
+    public void testScheduleWriteOperationWithNonAtomicRecords() throws ExecutionException, InterruptedException, TimeoutException {
```

Comment:
for my understanding, all existing tests use atomic records -- are there any that should be non-atomic?

Reply:
I think that the two new tests cover the non-atomic write. All the existing one must be kept as they are.